File: matlab/fsatellite.m
function [h,v] = fsatellite(T,unit)
% FSATELLITE is the functional form of the satellite script. It takes the
% orbital period of a satellite T and the units of time (optional). Units
% of time can include "hr" for hours (default), "min" for minutes, "s" for
% seconds, and "dy" for days. The function returns the altitude in km and
% velocity in km/s
 
if nargin==1 % nargin stores the number of arguments put in with the function
    unit= "hr";
elseif nargin ==0 
    error("Invalid input")    
end
% Defining constants
mts = 60;
hts= 3600;
dts = 86400;
G = 6.67*10^(-11);
M = 5.98*10^(24);
R = 6.371*10^(6);
% Calculations using anonymous functions!!
% Is this overkill? Maybe
V = @(r) sqrt(G*M./r);
H = @(t) (G*M*t.^2./(4*pi.^2)).^(1/3) - R;
% Checks the unit input
switch unit
    case "sec"
        t=T;
        tconvert = 1;
        lab = "sec";
    case "min"
        t= mts*T;
        tconvert = mts;
        lab = "min";
    case "hr" 
        t = hts*T; 
        tconvert = hts;
        lab = "hrs";
    case "dy"
        t = dts*T;
        tconvert = dts
        lab = "days";
    otherwise
        error("Invalid input")
end
% Calculate altitude and velocity
h = mtokm(H(t));
v = mtokm(V(R+H(t)));
time = 5200:100:8*t;
height = 0:100:H(time(end));
% Graph results in 2 figures
figure
plot(time/tconvert,mtokm(H(time)),'b-',t/tconvert,h,'r*')
ylabel("altitude (km)")
xlab = "period of orbit " + "(in " + lab + ")";
xlabel(xlab)
title('Altitude of Orbit')
figure
plot(mtokm(height),mtokm(V(height+R)),'b-',h,v,'r*')
xlabel('altitude (km)')
ylabel('velocity in orbit (km/s)')
title('Orbital Velocity')

end

function H = mtokm(h)
     % Converts input from meters to kilometers
     H = h./1000;
end
function h = kmtom(H) 
     % Converts input from kilometers to meters
     h = 1000.*H;
end

File: python/fsatellite.py
# -*- coding: utf-8 -*-
"""
This file contains functions related to calculating orbital quantities of satellites
"""
import numpy as np
import matplotlib.pyplot as pyp
from colorama import Fore, Back
G = 6.67*10**(-11);
M = 5.98*10**(24);
R = 6.371*10**(6);


def satellite(T,grph = 0,un='hr'):
    if un=="hr" or un=="hour":
        t = 3600*T;
    elif un == "day" or un == "dy":
        t = 24*3600*T;
    elif un == "min" or un== "m":
        t = 60*T;
    elif un == "s" or un=="sec":
        t=T;
    else:
        print(Fore.RED + "Error: need numerical input with a units of sec, min, hour or day")
        
    mtokm = lambda x: x/1000; #lambda functions are python's anonymous functions

    h= alt(t);
    v =orbvel(h);
    V = mtokm(v);
    H = mtokm(h);
    
    T = 3600*np.arange(1.5,60)
    h = alt(T);
    v = mtokm(orbvel(h));
    h = mtokm(h);
    if grph:
        fig(T,h);
        fig(v,h,'orbital velocity (km/s)');
    return H, V

def alt(t):
     h = (G*M*t**2/(4*np.pi**2))**(1/3) - R;
     return h

def orbvel(h):
    v = np.sqrt(G*M/(h+R));
    return v

def fig(x,y,xlab='period of orbit (hours)',ylab='altitude (km)'):
    pyp.figure()
    pyp.plot(x,y)
    pyp.xlabel(xlab)
    pyp.ylabel(ylab)
    
